src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -40,92 +44,79 @@ Hamiltonian, compressing shared interaction channels.
 
-  # traverse tree outwards from root vertex
-  vs = _default_vertex_ordering(sites, root_vertex)
-  # TODO: Add check in ttn_svd that the ordering matches that of find_index_in_tree, which is used in sorteachterm #fermion-sign!
-  # store edges in fixed ordering relative to root
-  es = _default_edge_ordering(sites, root_vertex)
-  # some things to keep track of
-  # rank of every TTN tensor in network
-  degrees = Dict(v => degree(sites, v) for v in vs)
-  # link isometries for SVD compression of TTN
-  Vs = Dict(e => Dict{QN,Matrix{coefficient_type}}() for e in es)
-  # map from term in Hamiltonian to incoming channel index for every edge
+  ordered_verts = _default_vertex_ordering(sites, root_vertex)
+  ordered_edges = _default_edge_ordering(sites, root_vertex)
   inmaps = Dict{Pair{edgetype_sites,QN},Dict{Vector{Op},Int}}()
-  # map from term in Hamiltonian to outgoing channel index for every edge
   outmaps = Dict{Pair{edgetype_sites,QN},Dict{Vector{Op},Int}}()
 
-  op_cache = Dict{Pair{String,vertextype_sites},ITensor}()
-
-  function calc_qn(term::Vector{Op})
-    q = QN()
-    for st in term
-      op_tensor = get(op_cache, ITensors.which_op(st) => ITensors.site(st), nothing)
-      if op_tensor === nothing
-        op_tensor = op(
-          sites[ITensors.site(st)], ITensors.which_op(st); ITensors.params(st)...
-        )
-        op_cache[ITensors.which_op(st) => ITensors.site(st)] = op_tensor
-      end
-      if !isnothing(flux(op_tensor))
-        q += flux(op_tensor)
-      end
-    end
-    return q
-  end
-
-  Hflux = -calc_qn(terms(first(terms(os))))
-  # insert dummy indices on internal vertices, these will not show up in the final tensor
-  is_internal = Dict{vertextype_sites,Bool}()
-  for v in vs
-    is_internal[v] = isempty(sites[v])
-    if isempty(sites[v])
-      # FIXME: This logic only works for trivial flux, breaks for nonzero flux
-      # TODO: add assert or fix and add test!
-      sites[v] = [Index(Hflux => 1)]
-    end
-  end
-  # bond coefficients for incoming edge channels
+  # Bond coefficients for incoming edge channels.
+  # These become the "M" coefficient matrices that get SVD'd.
   inbond_coefs = Dict(
-    e => Dict{QN,Vector{ITensorMPS.MatElem{coefficient_type}}}() for e in es
+    e => Dict{QN,Vector{MatElem{coefficient_type}}}() for e in ordered_edges
   )
-  # list of terms for which the coefficient has been added to a site factor
+
+  # List of terms for which the coefficient has been added to a site factor
   site_coef_done = Prod{Op}[]
-  # temporary symbolic representation of TTN Hamiltonian
-  tempTTN = Dict(v => QNArrElem{Scaled{coefficient_type,Prod{Op}},degrees[v]}[] for v in vs)
 
-  # build compressed finite state machine representation
-  for v in vs
-    # for every vertex, find all edges that contain this vertex
-    edges = align_and_reorder_edges(incident_edges(sites, v), es)
+  # Temporary symbolic representation of TTN Hamiltonian
+  tempTTN = Dict(
```

Comment:
```suggestion
  temp_ttn = Dict(
```

Reply:
Seems like this data structure could be a `DataGraph` with a graph structure matching the `IndsNetwork`/`TTN` graph structure and a `SparseArrayDOK` stored on the vertices, where the number of dimensions is the degree of the graph and the elements are `Scaled{coefficient_type,Prod{Op}}`. Does that sound right to you?

I suppose one thing that needs to be stored is the meaning of each dimension of the `SparseArrayDOK` on the vertices since you want to know which dimension corresponds to which neighbor. So interestingly the best representation may be an `ITensor`, or maybe a [NamedDimsArray](https://github.com/ITensor/ITensors.jl/tree/main/NDTensors/src/lib/NamedDimsArrays) wrapping a `SparseArrayDOK`, where the dimension names are the edges of the graph.